### ISSUE TITLE
can't attached to existing ID created on phone + problem with recovery phrase

### ISSUE THREAD

**Issue by boistordu** [loki-project/session-desktop#942]
**Describe the bug**
1/ I can't link a computer to one of my phones existing ID, it goes in an endless loop when I enter the session ID and there is no notification on the phone
2/ the recovery phrase is ending up with me getting another session ID than on my phone
**To Reproduce**
what is described in the description, no specific steps
**Screenshots or Logs**
If applicable, add screenshots or log files to help explain your problem.
self explenatory
**Other information (please complete the following information):**

* Device: [e.g. PC, Mac] PC
* OS: [e.g. Ubuntu 16.04, Windows 10] windows 10
* Session Version or Git commit hash: 1.0.2

**Comment by KeeJef** [comment]
Do you have any information about your primary device, Android or iOS and what version of Session is it using?

**Comment by boistordu** [comment]
the session ID has been created on iOS device just 24 hours ago 
The version is the last one available on my app store 1.0.5 apparently.

**Comment by RyanRory** [comment]
When you linked your desktop to your phone, was your phone on the device linking page with the dialog of the QRcode shown like this?
![image](https://user-images.githubusercontent.com/8543570/75742551-1644b080-5d62-11ea-8ac8-776bf1d7d419.png)

**Comment by boistordu** [comment]
yup that's it. 
I never got any notification even with just the screen before about something like "a device wants to link with your session" or something alike which I should have received...

**Comment by Hoimar** [comment]
I'm affected by this too. Created an account on Session Android 1.0.5, then wanted to link it using Session Desktop 1.0.2 following the required steps exactly, but I never get a notification on mobile, and the desktop app keeps showing me a loading bar.

**Comment by Niek** [comment]
Same problem here, see also #943

**Comment by alhamz** [comment]
Same problem, started original session on Android 10, tried linking on Ubuntu and the linking never completes even though it displays a recovery phrase for this device.

**Comment by Mikunj** [comment]
Could you try linking again and then go to Window > Debug log and submit the log and put the link here so we can get more information on where the problem is occurring? I can't seem to recreate this.

**Comment by boistordu** [comment]
@Mikunj I've retried. It seems to work again now with only the session ID, I didn't try the recovery phrase so that might still be a current bug.... 
I guess you want me to try it again?
And also I guess that you had problems of connectivity amongst your nodes. Did you patch or restarted some of them?

**Comment by Niek** [comment]
I can also confirm it's working now (I guess after updating the Android client).

**Comment by boistordu** [comment]
the unlink is working properly.
the recovery phrase is not working properly, it still creates a new session Id and so a new account... 

Actually when I linked with a session ID it gave me another recovery phrase.
Sorry but maybe I'm mad but for me that was not the purpose of a recovery phrase. A recovery phrase was to retrieve access to an account. Not to create a new one.... So maybe it has a different meaning for you...
Anyway 2 devices, same session ID and 2 different recovery phrases..

Also When linking to my ID, in paired devices on the new devices it doesn't show the one to I had paired with... I think it should show both ways...

**Comment by boistordu** [comment]
I forgot, here are 2 debug logs:
https://debuglogs.org/da76cb0e4fbd0b236a0fd0f6daf1b29d6ce93be7c15a1b6262b9e08bbd3689e3

https://debuglogs.org/572919fb9e288ed73a4542d5706589799433961942bf03da5d9b6c61611588d9

**Comment by neuroscr** [comment]
>And also I guess that you had problems of connectivity amongst your nodes. Did you patch or restarted some of them?

We don't run the servers. Our service node operators do. We have multiple redundancies per swarm, so we or our service node operators don't need to restart anything. The operators usually do a good job handling any problems.

> Also When linking to my ID, in paired devices on the new devices it doesn't show the one to I had paired with... I think it should show both ways...

Yes, of course, just a bug right now.

> I forgot, here are 2 debug logs:
Those are for 1.0.2, we just released 1.0.4 tonight that has recovery phrase fixes, I suggest you upgrade and try it and let us know

**Comment by boistordu** [comment]
okey thanks for the update @neuroscr 
But so if it was not a problem of connectivity since you have so much redundancies, then what could it be?

**Comment by neuroscr** [comment]
>what could it be?

most likely a bug. Android code base is different than the Desktop, so we have to continually test to make sure both are working correctly. There seems to be a few bugs left in getting the systems to talk to each other. If you can help provide us info like logs, that's a big help for tracking these issues down.

**Comment by boistordu** [comment]
okey. Mine was IOS. so between windows and IOS same diagnosis?
There is no updater inside the app I guess yet? on windows I mean

**Comment by neuroscr** [comment]
We just added the auto-updater in 1.0.3, so newer releases will have it

**Comment by boistordu** [comment]
thanks for all the help :)

**Comment by Hoimar** [comment]
Linking devices works for me now too, even without updating both components, Session Android and Session Desktop for Linux (I still updated them now).

**Comment by CrafterLaughter** [comment]
With the latest versions as of 3-17 I tested linking iPhone to Mac.  With either iphone as primary and Mac as secondary or the inverse what happens is the secondary device presents "here is your secret" but the secret confirmation dialogue never reaches the primary device.  So unfortunately it doesn't work in either direction.

**Comment by CrafterLaughter** [comment]
Then I moved on to test my Win10PC with Android (latest versions of Session as of 3-17).  They linked successfully (desktop primary, android secondary).  I fresh installed Session on the iphone again and sent a message to Win10PC.  Success.  When I sent a reply message to the iphone it delivered to the linked android handset instead.  Super weird.  The Win10PC is 1909 build 18363.720 and Android is 10 March 4 update.

I anticipated that linking devices would operate like Signal with keeping Win10PC and Android messages threads synced as if controlling a single account.

Both the Win10PC and Android show the identical session ID keystring and show as linked under "devices" in the GUI.
